src/app/api/api_v1/endpoints/detections.py

```diff
@@ -64,45 +64,77 @@ async def create_detection(
     detections: DetectionCRUD = Depends(get_detection_crud),
     token_payload: TokenPayload = Security(get_jwt, scopes=[UserRole.ADMIN, UserRole.AGENT, UserRole.USER]),
 ) -> Detection:
     telemetry_client.capture(token_payload.sub, event="detections-get", properties={"detection_id": detection_id})
-    return cast(Detection, await detections.get(detection_id, strict=True))
+    detection = cast(Detection, await detections.get(detection_id, strict=True))
+
+    if UserRole.ADMIN in token_payload.scopes:
+        return detection
+
+    camera = cast(Camera, await cameras.get(detection.camera_id, strict=True))
+    if token_payload.organization_id != camera.organization_id and UserRole.ADMIN not in token_payload.scopes:
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="Access forbidden.")
+    return detection
 
 
 @router.get("/{detection_id}/url", response_model=DetectionUrl, status_code=200)
 async def get_detection_url(
     detection_id: int = Path(..., gt=0),
+    cameras: CameraCRUD = Depends(get_camera_crud),
     detections: DetectionCRUD = Depends(get_detection_crud),
     token_payload: TokenPayload = Security(get_jwt, scopes=[UserRole.ADMIN, UserRole.AGENT, UserRole.USER]),
 ) -> DetectionUrl:
     """Resolve the temporary media image URL"""
     telemetry_client.capture(token_payload.sub, event="detections-url", properties={"detection_id": detection_id})
     # Check in DB
     detection = cast(Detection, await detections.get(detection_id, strict=True))
+
+    if UserRole.ADMIN in token_payload.scopes:
+        return DetectionUrl(url=await s3_bucket.get_public_url(detection.bucket_key))
+
+    camera = cast(Camera, await cameras.get(detection.camera_id, strict=True))
+    if token_payload.organization_id != camera.organization_id:
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="Access forbidden.")
     # Check in bucket
-    temp_public_url = await s3_bucket.get_public_url(detection.bucket_key)
-    return DetectionUrl(url=temp_public_url)
+    return DetectionUrl(url=await s3_bucket.get_public_url(detection.bucket_key))
 
 
 @router.get("/", status_code=status.HTTP_200_OK, summary="Fetch all the detections")
 async def fetch_detections(
     detections: DetectionCRUD = Depends(get_detection_crud),
+    cameras: CameraCRUD = Depends(get_camera_crud),
     token_payload: TokenPayload = Security(get_jwt, scopes=[UserRole.ADMIN, UserRole.AGENT, UserRole.USER]),
 ) -> List[Detection]:
     telemetry_client.capture(token_payload.sub, event="detections-fetch")
-    return [elt for elt in await detections.fetch_all()]
+    if UserRole.ADMIN in token_payload.scopes:
+        return [elt for elt in await detections.fetch_all()]
+
+    cameras_list = await cameras.get_all_by("organization_id", token_payload.organization_id, strict=True)
+    camera_ids = [camera.id for camera in cameras_list]
+
+    return await detections.get_in(camera_ids, "camera_id")
```

Comment:
Perfect for the SQL ops. One feedback though: let's improve those API
- `get_all_by` --> you already have `fetch_all` using the `filter_pair` argument
```python
cameras_list = await cameras.fetch_all(("organization_id", token_payload.organization_id))
```
- `get_in`  --> I suggest extending `fetch_all` instead with a new arg "" to allow in filter, or by extending "filter_pair" to have either `Tuple[str, scalar]` or `Tuple[str, Sequence[scalar]]` type (in second case, do "in" operation rather than "where". To get something like:
```python
return await detections.fetch_all(("camera_id", camera_ids))
```

Reply:
done in the Acknowledged PR